### PR TITLE
fix: stop global setup when debugger is terminated

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/microsoft/playwright/issues/32387

We're using the cancellation token passed by VS Code to abort an ongoing debug run. There's two problems with this:

1. VS Code doesn't cancel the token if the debugger is terminated
1. If the debugger terminates during `runGlobalSetup`, the underlying process will go away and stall `runGlobalSetup`.

This PR fixes both by adding the debugger termination onto the cancellation token and racing `runGlobalSetup` against it.

### CODE REVIEW

File: src/playwrightTestServer.ts [hunk 1]
```diff
@@ -255,11 +255,12 @@ export class PlaywrightTestServer {
       if (!locations && !testIds)
         return;
 
-      const result = await this._runGlobalHooksInServer(debugTestServer, 'setup', reporter);
+      const result = await Promise.race([
+        this._runGlobalHooksInServer(debugTestServer, 'setup', reporter),
```
Comment (dgozman) [inline src/playwrightTestServer.ts:259]: I think we should pass the cancellation token to `_runGlobalHooksInServer()` instead, and make sure to pass it from any other callsites.

**Comment by dgozman** [comment]
The change looks good, but bots are unhappy. Do you think we can add a test for this?

**Comment by Skn0tt** [comment]
Fixed the bots.  https://github.com/microsoft/playwright-vscode/pull/525/commits/7f02bbbd51b961a593a043621a50dada3bdcde7b adds a test that fails on `main` and passes on this branch.

**Comment by Skn0tt** [comment]
Ha, turns out there was another bug inside the legacy route!

**Comment by Skn0tt** [comment]
@dgozman could you take another look?

**Review by dgozman — APPROVED** [review]
Looks good!

**Pull request merged**